src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -300,10 +311,12 @@ class MarathonSchedulerService @Inject() (
   }
 
   def abdicateLeadership(): Unit = {
-    log.info("Abdicating")
+    if (leader.get()) {
```

Comment:
hmmm, that probably happens elsewhere but I am suspicious of using an atomic guard to prevent duplicated execution and not using an atomic operation that also sets this flag to false.